app/src/main/java/edu/usf/sas/pal/muser/util/HeadphoneUtils.kt

```diff
@@ -0,0 +1,111 @@
+package edu.usf.sas.pal.muser.util
+
+import android.content.Context
+import android.media.AudioDeviceInfo
+import android.media.AudioManager
+import android.os.Build
+import android.util.Log
+import edu.usf.sas.pal.muser.model.AudioData
+import edu.usf.sas.pal.muser.model.AudioDeviceType
+import edu.usf.sas.pal.muser.model.VolumeData
+
+object HeadphoneUtils {
+
+    private const val TAG = "HeadphoneUtils"
+    private var deviceType = 0
+
+    /**
+     * Function to populate volume data class.
+     * @param context - context of the activity/fragment.
+     * @return VolumeData object.
+     */
+    @JvmStatic
+     fun getVolumeData(context: Context): VolumeData? {
+         val audioManager = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager?
+         if (audioManager != null) {
+             val volumeLevel = audioManager.getStreamVolume(AudioManager.STREAM_MUSIC)
+             val volumeMax = audioManager.getStreamMaxVolume(AudioManager.STREAM_MUSIC)
+             var volumeMin = Int.MIN_VALUE
+             var volumeLevelDB = Float.MIN_VALUE
+             if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                 volumeMin = audioManager.getStreamMinVolume(AudioManager.STREAM_MUSIC)
+                 volumeLevelDB = if (isSpeakerOn(context)) {
+                     audioManager.getStreamVolumeDb(AudioManager.STREAM_MUSIC,
+                             volumeLevel, AudioDeviceInfo.TYPE_BUILTIN_SPEAKER)
+                 } else {
+                     audioManager.getStreamVolumeDb(AudioManager.STREAM_MUSIC,
+                             volumeLevel, deviceType)
+                 }
+             }
+             return VolumeData(volumeLevel, volumeMax, volumeMin, volumeLevelDB)
+         }
+        return null
+     }
+
+    /**
+     * Function to check if the built in speaker is used or not
+     * @param context - context of the activity/fragment
+     * @return isSpeakerOn true or false.
+     */
+    @JvmStatic
+    fun isSpeakerOn(context: Context): Boolean{
```

Comment:
If any of these methods are only accessed within this class they should be `private`

Reply:
If returning the `Pair` here I would rename this to `getAudioDevices()`.